Tryouts/Core/Abstractions/ISubscriptionClient.cs

```diff
@@ -18,10 +18,13 @@ public interface ISubscriptionClient
     {
         Task Connect(string host, int port);
 
-        Task Publish<TMessage>(Topic topic, TMessage msg)
-             where TMessage : IMessage;
+        Task Publish(Topic topic, IMessage msg);
```

Comment:
`IMessage` is a gRPC-specific type. This is against every requirement we set for the Message Router so far - messages should be plain text or raw bytes and the communication protocol should not leak into the API.
Topics are still defined as an enumeration in an abstractions package - application should be able to use arbitrary topics.

Reply:
I completely changed ISubscriptionClient to be more in tune with Message router API:

    public interface ISubscriptionClient
    {
        Task Connect(string host, int port);

        Task Publish(string topic, string msg);

        (IObservable<CommunicationsMessage> messageStream, IDisposable unsubscriber)
            Subscribe(string topic);
    }

Published message is now string, topic is a string and Subscribe returns `CommunicationsMessage` that is exactly the same as `RoutedMessage` (once we integrate, I'll use `RoutedMessage`)
Please, see if this is ok with you.